src/DSharp.Compiler/Generator/ExpressionGenerator.cs

```diff
@@ -541,6 +524,34 @@ private static void GenerateObjectExpression(ScriptGenerator generator, MemberSy
             writer.Write("}");
         }
 
+        public static void GenerateObjectInitializerExpression(
+            ScriptGenerator generator, 
+            MemberSymbol symbol, 
+            ObjectInitializerExpression initializerExpression)
+        {
+            ScriptTextWriter writer = generator.Writer;
+
+            var type = initializerExpression.NewExpression.EvaluatedType;
+
+            if(type.IsReservedType())
+            {
+                //TODO: Implement Dictionary initializer + List Initializer + Custom initializers
+                GenerateNewExpression(generator, symbol, initializerExpression.NewExpression);
```

Comment:
collection initializers in C# work by convention. So if the object implements an ICollection then you should be able to use an initializer cause it will compile (in IL) to multiple calls to `.Add(T)`. We could do exactly the same

Reply:
Yeh I was reviewing the specification yesterday, it'd be fairly easy to use this pattern. I'd have to write a piece of code that can initialize lists and dictionary based implementations. Should be fun!